src/harness/reference_models/fss_purge/fss_purge.py

```diff
@@ -0,0 +1,332 @@
+    cbsds: List of CbsdData objects that are in the neighbourhood to an FSS.
+    fss_entity: The FssProtectionPoint tuple of the FSS for which the purge is to be performed.
+    fss_oobe_margin: OOBE Margin value to be considered for FSS purge list calculation.
+  """
+
+  # Calculate the OOBE interference value for each CBSD based on the grant with the highest frequency.
+  calculateOobeInterference(cbsds, fss_entity)
+
+  # Sort the CBSDs based on the calculated interference value.
+  sorted_cbsds = sorted(cbsds, key=lambda cbsd: cbsd['oobe_interference'])
+  # Calculate the threshold value in dBm for the OOBE interference
+  oobe_threshold_value = -129 + 10 * (np.log10(REF_BW/ONE_MHZ) - fss_oobe_margin)
+
+  # Find the largest index such that aggregation of the interferences do not cross 
+  # the OOBE threshold
+  oobe_interferences = [db2Linear(cbsd['oobe_interference']) for cbsd in sorted_cbsds]
+  cumulated_interference = np.cumsum(oobe_interferences)
+  index = np.searchsorted(cumulated_interference, db2Linear(oobe_threshold_value))
+  cbsds_to_purge = sorted_cbsds[index:]
+  sorted_cbsds = sorted_cbsds[:index]
+
+  # Remove the grants from the CBSDs that are in the purge list.
+  if cbsds_to_purge:
+    for index, cbsd in enumerate(cbsds_to_purge):
+      # Get the grant that was used for the OOBE interference calculation, that
+      # is the grant with highest highFrequency.
+      grant_with_max_high_freq = max(cbsd['grantRequests'],\
+            key=lambda x: x['operationParam']['operationFrequencyRange']['highFrequency'])
+
+      # Get all the grants that share the same emission mask as the one that 
+      # was used for OOBE calculation
+      if (grant_with_max_high_freq['operationParam']['operationFrequencyRange']
+          ['highFrequency'] > 3690000000):
+
+        # Adding the -13 emission mask grants to the purge list 
+        
+        grants_to_purge = [grant_request for grant_request in cbsd['grantRequests'] 
+                            if grant_request['operationParam']['operationFrequencyRange']
+                            ['highFrequency'] > 3690000000]
+      else:
+        # Adding the -25 emission mask grants to the purge list
+        grants_to_purge = [grant_request for grant_request in cbsd['grantRequests'] 
+                            if grant_request['operationParam']['operationFrequencyRange']
+                            ['highFrequency'] <= 3690000000]
+
+      # Removing grant requests that are in the purge list from the CBSDs.
+      for grant_items in grants_to_purge:
+         cbsd['grantRequests'].remove(grant_items)
+
+      # If no grants are remaining then remove the CBSD.
+      if not cbsd['grantRequests']:
+         cbsds_to_purge.remove(cbsd)
+      else:
+         sorted_cbsds.append(cbsd)
+
+    performPurge(sorted_cbsds, fss_entity, fss_oobe_margin)
+    
+  # Removing the oobe_interference parameter added to cbsd objects
+  for cbsd in cbsds:
+    cbsd.pop('oobe_interference', None)

```

Comment:
You can use `del` instead since you don't need to retrieve the value.

Reply:
Updated   with del function